client/my-sites/themes/controller.jsx

```diff
@@ -43,6 +44,13 @@ export function loggedOut( context, next ) {
 		return next();
 	}
 
+	// If logged in, do nothing:
+	const state = context.store.getState();
+	const currentUser = getCurrentUser( state );
```

Comment:
We also have a specialized `isUserLoggedIn( state )` selector that returns a boolean.

Reply:
This extra condition is not needed if the `selectSiteIfLoggedIn` handler terminates early and doesn't call `next`.

One way to think about it is that earlier handlers in the chain check pre-conditions for later handlers and proceed only if the pre-conditions are satisfied.